docs: update utility fn docs (#13310)
AztecProtocol/aztec-packages: docs/docs/aztec/concepts/pxe/acir_simulator.md

Public functions can call other public functions as well as private functions. Public function composability can happen atomically, but public to private function calls happen asynchronously (the private function call must happen in a future block).

### Utility Functions

> Unconstrained functions are used to extract useful data for users, such as the user balance. They are not proven, and are simulated client-side.

Reviewer: ```suggestion
Utility functions are used to extract useful data for users, such as the user balance. They are not proven, and are simulated client-side.
```

> They are run with the assistance of a DB oracle that provides any private data requested by the function.

Reviewer: ```suggestion
They are run with the assistance of an [oracle resolver](https://noir-lang.org/docs/explainers/explainer-oracle) that provides any private data requested by the function.
```
DB
